aicsimageio/readers/ome_zarr_reader.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import xarray as xr
+from fsspec.implementations.local import LocalFileSystem
+from fsspec.spec import AbstractFileSystem
+
+from .. import exceptions, types
+from ..dimensions import DimensionNames
+from ..metadata import utils as metadata_utils
+from ..utils import io_utils
+from .reader import Reader
+
+try:
+    from ome_zarr.io import parse_url
+    from ome_zarr.reader import Reader as ZarrReader
+
+except ImportError:
+    raise ImportError(
+        "ome_zarr is required for this reader. " "Install with `pip install 'ome_zarr'`"
+    )
+
+###############################################################################
+
+
+class OmeZarrReader(Reader):
+    """
+    Wraps the ome-zarr-py API to provide the same aicsimageio Reader API but for
+    OmeZarr images.
+
+    Parameters
+    ----------
+    image: types.PathLike
+        Path to image file to construct Reader for.
+    fs_kwargs: Dict[str, Any]
+        Any specific keyword arguments to pass down to the fsspec created filesystem.
+        Default: {}
+    """
+
+    @staticmethod
+    def _is_supported_image(fs: AbstractFileSystem, path: str, **kwargs: Any) -> bool:
+        try:
+            _ = fs  # TODO: How to use the FS to test for suppored image
+            ZarrReader(parse_url(path, mode="r"))
+            return True
+
+        except ValueError:
+            return False
+
+    def __init__(
+        self,
+        image: types.PathLike,
+        channel_names: Optional[Union[List[str], List[List[str]]]] = None,
+        fs_kwargs: Dict[str, Any] = {},
+    ):
+        # Expand details of provided image
+        self._fs, self._path = io_utils.pathlike_to_fs(
+            image,
+            enforce_exists=True,
+            fs_kwargs=fs_kwargs,
+        )
+
+        if not isinstance(self._fs, LocalFileSystem):
+            raise ValueError(
+                f"Cannot read OmeZarr from non-local file system. "
+                f"Received URI: {self._path}, which points to {type(self._fs)}."
+            )
+
+        # Enforce valid image
+        if not self._is_supported_image(self._fs, self._path):
+            raise exceptions.UnsupportedFileFormatError(
+                self.__class__.__name__, self._path
+            )
+
+        self._channel_names = channel_names
```

Comment:
Why would a user want to supply channel names themselves?

Reply:
removed with [9c2ccf6](https://github.com/AllenCellModeling/aicsimageio/pull/493/commits/9c2ccf6506f33867fc773355c52a38d6c01b6c0e)